src/libDirectoryService/Coinbase.cpp

```diff
@@ -158,10 +188,38 @@ void DirectoryService::InitCoinbase() {
 
       for (auto const& addr : shardId.second) {
         if (!AccountStore::GetInstance().UpdateCoinbaseTemp(
-                addr, genesisAccount, COINBASE_REWARD)) {
+                addr, genesisAccount, reward_each)) {
           LOG_GENERAL(WARNING, "Could Not reward " << addr);
+        } else {
+          suc_counter++;
         }
       }
     }
   }
+
+  uint256_t balance_left = total_reward - (suc_counter * reward_each);
```

Comment:
Can use safemath?

Reply:
safe math was already fully covered for the context of this calc, otherwise it's redundant